ui/js/controllers/main.js

```diff
@@ -582,8 +582,15 @@ treeherderApp.controller('MainCtrl', [
         $scope.fieldChoices = thJobFilters.getFieldChoices();
 
         $scope.startNewFieldFilter = function () {
+            // provide toggle behavior on the filter by job field btn
+            if ($scope.newFieldFilter !== null) {
+                if ($scope.newFieldFilter.field === "" && $scope.newFieldFilter.value === "") {
+                    $scope.cancelNewFieldFilter();
+                } return;
```

Comment:
Wouldn't this ``return`` normally be on the next line?  Other than that, this looks great!  Though, would you add a short comment stating that if the user has modified the values of the ``newFieldFilter`` that this won't kill it?  Fairly obvious, but a comment won't hurt.  :)

Reply:
Thanks @camd! I've been looking at the UI behaviour again just now, and I think I'd like to do a `toggleFieldFilterVisibility()` sort of thing; so if I close and re-open the filter bar via the button, the dropdown choices will be preserved. Whereas the Cancel button will clear the choices. I'm mucking away on that right now.